### PR TITLE
Keyword to keyid

### PR DESCRIPTION
**Pull request by Mariana-21**
## Spec/Features ##
* Added VCR to specs
* Changed server from local to Heroku

## Routes ## 
* Get route for /game_search index method

## Controllers ## 
* Changed index method for game_search
* Added to create method in game_search controller to get games from keywords and fear quiz

## Poros ## 
* Added new method to get games by keywords

## Views ##
* Took out some functionality in quiz.scss to all checkbox tag 
* Changed quiz to be in a form

### CODE REVIEW

File: app/controllers/game_search_controller.rb [hunk 1]
```diff
@@ -36,6 +29,16 @@ def create_game_objects(params)
...
+    elsif params[:search_type] == 'keyword'
+      games = results.games_by_keywords(params[:search])
+      return games if games.is_a?(String)
+      session[:search_ids] = games.map(&:id) if games.is_a?(Array)
+    elsif params[:search_type] == 'quiz'
+      boxes = params.keys.find_all {|param| param.include?("box")}
+      fear_keywords = params.permit(boxes).to_h.values.join(',')
+      
+      games = results.games_by_keywords(fear_keywords)
+      session[:search_ids] = games.map(&:id) if games.is_a?(Array)
```
Comment (perryr16) [inline app/controllers/game_search_controller.rb:41]: were thinking this is getting a bit crowded and convoluted. A refactor to multiple controllers like `game_game_search_controller`
`keyword_game_search_controller`
`quiz_game_search_controller`

...but that's a lot of work and maybe something to talk about in the eval for areas to improve

File: app/poros/i_scream_results.rb [hunk 2]
```diff
@@ -112,4 +112,22 @@ def new_game_params(data)
...
   end
+
+  def games_by_keywords(keywords)
+    results = service.get_games_by_keywords(keywords)
+    return results if results.is_a?(String)
+    results[:data].map do |api_return|
+      data = api_return[:data]
+      game = game_object(data)
+
+      category_objects(data, game)
```
Comment (perryr16) [inline app/poros/i_scream_results.rb:123]: Each one of these methods generates the model in the title, see below for example

File: app/services/i_scream_service.rb [hunk 3]
```diff
@@ -1,8 +1,12 @@
 class IScreamService 
 
   def conn 
-    # Faraday.new('http://localhost:9393')
-    Faraday.new('https://i-scream-microservice.herokuapp.com')
+    # Faraday.new('http://localhost:9393') do |faraday|
+
+    Faraday.new('https://i-scream-microservice.herokuapp.com') do |faraday|
+      faraday.options[:timeout] = 300
```
Comment (perryr16) [inline app/services/i_scream_service.rb:7]: API is so slow we had to increase the timeout settings :/

File: spec/features/game_search/welcome_spec.rb [hunk 4]
```diff
@@ -39,6 +39,14 @@
     expect(page).to have_content("Invalid Game Name")
     
   end
+
+  xit "can bring back games based off of a keyword" do
```
Comment (perryr16) [inline spec/features/game_search/welcome_spec.rb:43]: needs some love

File: app/assets/stylesheets/quiz.scss [hunk 5]
```diff
@@ -1,8 +1,3 @@
-/* Hide the Ordinary Checkbox */
-input[type="checkbox"] {
```
Comment (melatran) [inline app/assets/stylesheets/quiz.scss:2]: Bye bye fancy checkbox

File: app/controllers/game_search_controller.rb (hunk 1 above)
Comment (melatran) [reply app/controllers/game_search_controller.rb:41]: To me, this makes sense. The controller names can get messy like game game sounds kind of strange. Maybe something using namespace and put different aspects that depend on game within that namespace. This sounds good to mention in our things to improve on

---> game
           ---> game:search
          ----> game:keyword

File: app/poros/i_scream_results.rb (hunk 2 above)
Comment (melatran) [reply app/poros/i_scream_results.rb:123]: This looks really nice

File: app/services/i_scream_service.rb (hunk 3 above)
Comment (melatran) [reply app/services/i_scream_service.rb:7]: :( but for our eval demo, we should just use the same searches and do what ian suggested and just cache it

**Pull request merged**